src/frontend/mosh-server.cc

```diff
@@ -592,8 +603,12 @@ static int run_server( const char *desired_ip, const char *desired_port,
 
     /* Wait for parent to release us. */
     char linebuf[81];
-    if (fgets(linebuf, sizeof linebuf, stdin) == NULL) {
-      err( 1, "parent signal" );
+    // -1, errno == EINTR -- retry
+    // otherwise, give up
+    while ( read( pipes[0], linebuf, sizeof( linebuf ) ) == -1 ) {
```

Comment:
Am I understanding correctly that before, we tried to call fgets() until we read a newline, then did execve, but now, we're going to do read() until the other end of the pipe is closed (i.e., EOF)? 

That means, AIUI, if we do get an error here and it's not EINTR, it's fatal?

Reply:
Prior to this change, I believe it was possible for the `fgets()` until we read a newline to block forever, in the case where the newline was never received. With this change, that hang at startup becomes a fatal error instead. The fatal error seems preferable to me, but I might be misunderstanding what the original code was doing.

Reply:
Discussed offline, this is not a behavior change, there was a fatal error on fgets hitting EOF before and there still is now.